### ISSUE TITLE
interpreter command not found

### ISSUE THREAD

**Issue by jordanbtucker** [KillianLucas/open-interpreter#169]
If you see an error similar to the following.

```
interpreter command not found
```

This is likely an issue with your Python/pip `Scripts` directory not being in your `PATH` environment variable.

To find out what that path is, run the following command.

```
python -c 'import os,sysconfig;print(sysconfig.get_path("scripts"))'
```

Now [add that directory](https://gist.github.com/nex3/c395b2f8fd4b02068be37c961301caa7#file-path-md) to your PATH environment variable and restart your terminal.

If you still run into problems, leave a comment here or ask on [the Discord server](https://discord.gg/YG7APUyJ5).

**Comment by veonua** [comment]
on my system

> print(sysconfig.get_path("scripts"))
C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.1520.0_x64__qbz5n2kfra8p0\Scripts
is missing and protected root folder

the right path is

> python -m site --user-site 
C:\Users\{user}\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\Scripts

**Comment by pyoneerC** [comment]
PATH or path? Only got the latter

**Comment by milgodlewski** [comment]
In my case, I had the correct path to "/scripts" in PATH during the installation process. 
Important to mention - I initially started the installation on Python 3.11.0
<img width="701" alt="image" src="https://github.com/KillianLucas/open-interpreter/assets/15853604/968adaf9-617c-4ad7-8897-690d4edf7388">

The installation was successfully completed on Python version 3.11.4

**Comment by ys112** [comment]
Hi, for anyone else that might face same issues where interpreter still fails to work after adding "C:\Python311\Scripts" to PATH in windows.
Turns out that `pip install open-interpreter` have to run in **elevated command** in windows if not the error below will appear on first install (which disappears on subsequent same pip install commands)

>   WARNING: Failed to write executable - trying to use .deleteme logic
> ERROR: Could not install packages due to an OSError: [WinError 2] The system cannot find the file specified: 'C:\\Python311\\Scripts\\interpreter.exe' -> 'C:\\Python311\\Scripts\\interpreter.exe.deleteme'

which indicates that interpreter.exe is not installed in the python scripts folder hence interpreter command won't work.

**Solution:**
Open command prompt with admin rights
If you ran `pip install open-interpreter` previously,
run `pip uninstall open-interpreter` first.
Then  `pip install open-interpreter` to install again and no error should show up.
Run `interpreter` and voila

**Comment by pdwytr** [comment]
You'll 3.11 version for this to work, thats what helped me out. @jordanbtucker.

**Comment by pyoneerC** [comment]
have 3.11.5 still dont work

**Comment by hamoudydev** [comment]
I've had this issue since yesterday and none of the fixes provided worked for me. When running `interpreter` it would give me the error that it could not open the file and it showed the directory with `\\` and not `\`. 

My fix was to manually go to the directory of where interpreter.exe is located and running it manually from there and that seems to work.

**Comment by pyoneerC** [comment]
where is it located?

**Comment by Raclez** [comment]
I did what you did
```
ryu@ryu-2 ~ % python3 -c 'import os,sysconfig;print(sysconfig.get_path("scripts"))'
/usr/local/bin


ryu@ryu-2 ~ % echo $PATH

/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/aria2/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin

ryu@ryu-2 ~ % interpreter -y
zsh: command not found: interpreter
``` 
But it still doesn't work. That's why my head hurts

**Comment by hamoudydev** [comment]
> where is it located?

If `interpreter` doesn't work, try `py interpreter` and it should spit out the directory.

**Comment by maxpetrusenko** [comment]
Install -conda and create a new env, then run enterpreter

**Comment by jordanbtucker** [comment]
> > where is it located?
> 
> If `interpreter` doesn't work, try `py interpreter` and it should spit out the directory.
> 
> For me the directory was `C:\Users\user\AppData\Local\Programs\Python\Python311\Scripts`

```
C:\Users\Jordan\scoop\apps\python\current\python.exe: can't open file 'C:\\Users\\Jordan\\interpreter': [Errno 2] No such file or directory
```

**Comment by pyoneerC** [comment]
Nah still don't work

**Comment by jordanbtucker** [comment]
If you run the following command, it should tell you where the `interpreter` executable is.

```
pip show --files open-interpreter
```

```
Name: open-interpreter
Version: 0.1.3
Summary: Let language models run code locally.
Home-page:
Author: Killian Lucas
Author-email: killian@drinkwater.ai
License:
Location: c:\users\jordan\scoop\persist\miniconda3\envs\oi\lib\site-packages
Requires: appdirs, astor, git-python, huggingface-hub, inquirer, litellm, openai, pyreadline3, python-dotenv, rich, six, tiktoken, tokentrim, wget
Required-by:
Files:
  ..\..\Scripts\interpreter.exe
  interpreter\README.md
  interpreter\__init__.py
  ...
```

The first path under the `Files:` list should be the `interpreter` executable. It should be relative to the path to the `pip` executable, which you can find with the following command.

```
# Windows
where.exe pip

# Linux
which pip
```

On my Windows computer, it returns two paths for `pip` because I have Miniconda installed. The first path listed is the one that the `interpreter` path is relative to.

```
C:\Users\Jordan\scoop\apps\miniconda3\current\envs\oi\Scripts\pip.exe
C:\Users\Jordan\scoop\apps\python\current\Scripts\pip.exe
```

So, resolving `..\..\Scripts\interpreter.exe` relative to `C:\Users\Jordan\scoop\apps\miniconda3\current\envs\oi\Scripts\pip.exe` gives me `C:\Users\Jordan\scoop\apps\miniconda3\current\envs\oi\Scripts`

So, in my case, I need to make sure that `C:\Users\Jordan\scoop\apps\miniconda3\current\envs\oi\Scripts` is in my `PATH` environment variable.

**Comment by d0rb** [comment]
I was unable to solve this, but I think the reason you were unable to run it via the CLI is simply because you installed Python from the Microsoft Store.

This script will print the script's folder , 
`import sys
import os
virtual_env_directory = sys.prefix
scripts_folder = os.path.join(virtual_env_directory, "Scripts")
print("Scripts folder:", scripts_folder)`

Mine results as "Scripts folder: C:\Program Files\**WindowsApps**\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\Scripts"

If you have a Windows app in the print, you had installed it from the store. I cannot access my WindowsApps folder due to my workplace policies, but I think this might be the cause.

**Comment by jordanbtucker** [comment]
@d0rb Good find! Best to install Python from its website or via scoop/choco/etc

**Comment by meloseven** [comment]
> If you run the following command, it should tell you where the `interpreter` executable is.
> 
> ```
> pip show --files open-interpreter
> ```
> 
> ```
> Name: open-interpreter
> Version: 0.1.3
> Summary: Let language models run code locally.
> Home-page:
> Author: Killian Lucas
> Author-email: killian@drinkwater.ai
> License:
> Location: c:\users\jordan\scoop\persist\miniconda3\envs\oi\lib\site-packages
> Requires: appdirs, astor, git-python, huggingface-hub, inquirer, litellm, openai, pyreadline3, python-dotenv, rich, six, tiktoken, tokentrim, wget
> Required-by:
> Files:
>   ..\..\Scripts\interpreter.exe
>   interpreter\README.md
>   interpreter\__init__.py
>   ...
> ```
> 
> The first path under the `Files:` list should be the `interpreter` executable. It should be relative to the path to the `pip` executable, which you can find with the following command.
> 
> ```
> # Windows
> where.exe pip
> 
> # Linux
> which pip
> ```
> 
> On my Windows computer, it returns two paths for `pip` because I have Miniconda installed. The first path listed is the one that the `interpreter` path is relative to.
> 
> ```
> C:\Users\Jordan\scoop\apps\miniconda3\current\envs\oi\Scripts\pip.exe
> C:\Users\Jordan\scoop\apps\python\current\Scripts\pip.exe
> ```
> 
> So, resolving `..\..\Scripts\interpreter.exe` relative to `C:\Users\Jordan\scoop\apps\miniconda3\current\envs\oi\Scripts\pip.exe` gives me `C:\Users\Jordan\scoop\apps\miniconda3\current\envs\oi\Scripts\interpreter.exe`
> 
> So, in my case, I need to make sure that `C:\Users\Jordan\scoop\apps\miniconda3\current\envs\oi\Scripts` is in my `PATH` environment variable.

thks, this works for me

**Comment by songyaolun** [comment]
> Hi, for anyone else that might face same issues where interpreter still fails to work after adding "C:\Python311\Scripts" to PATH in windows. Turns out that `pip install open-interpreter` have to run in **elevated command** in windows if not the error below will appear on first install (which disappears on subsequent same pip install commands)
> 
> > WARNING: Failed to write executable - trying to use .deleteme logic
> > ERROR: Could not install packages due to an OSError: [WinError 2] The system cannot find the file specified: 'C:\Python311\Scripts\interpreter.exe' -> 'C:\Python311\Scripts\interpreter.exe.deleteme'
> 
> which indicates that interpreter.exe is not installed in the python scripts folder hence interpreter command won't work.
> 
> **Solution:** Open command prompt with admin rights If you ran `pip install open-interpreter` previously, run `pip uninstall open-interpreter` first. Then `pip install open-interpreter` to install again and no error should show up. Run `interpreter` and voila
thks, this one really did work for me.

**Comment by songyaolun** [comment]
@ys112 thks, this one really did work for me.

**Comment by arm5472** [comment]
I had the same issue using PowerShell 7, but it worked fine, using Command Prompt.

**Comment by afroze9** [comment]
After adding the scripts folder to your PATH, try running `interpreter.exe`. I'm running powershell 7 under windows terminal and it was unable to recognize the simple `interpreter` command.

**Comment by 1qzxc** [comment]
> Hi, for anyone else that might face same issues where interpreter still fails to work after adding "C:\Python311\Scripts" to PATH in windows. Turns out that `pip install open-interpreter` have to run in **elevated command** in windows if not the error below will appear on first install (which disappears on subsequent same pip install commands)
> 
> > WARNING: Failed to write executable - trying to use .deleteme logic
> > ERROR: Could not install packages due to an OSError: [WinError 2] The system cannot find the file specified: 'C:\Python311\Scripts\interpreter.exe' -> 'C:\Python311\Scripts\interpreter.exe.deleteme'
> 
> which indicates that interpreter.exe is not installed in the python scripts folder hence interpreter command won't work.
> 
> **Solution:** Open command prompt with admin rights If you ran `pip install open-interpreter` previously, run `pip uninstall open-interpreter` first. Then `pip install open-interpreter` to install again and no error should show up. Run `interpreter` and voila

I just wanted to point out that 'sudo python3.10 -m pip install open-interpreter' also fixed this for me on Mac M1

**Comment by MikeBirdTech** [comment]
This fix has been documented. Please create a new issue if the problem persists for you. Thanks!

**Comment by nothingness6** [comment]
> Hi, for anyone else that might face same issues where interpreter still fails to work after adding "C:\Python311\Scripts" to PATH in windows. Turns out that `pip install open-interpreter` have to run in **elevated command** in windows if not the error below will appear on first install (which disappears on subsequent same pip install commands)
> 
> > WARNING: Failed to write executable - trying to use .deleteme logic
> > ERROR: Could not install packages due to an OSError: [WinError 2] The system cannot find the file specified: 'C:\Python311\Scripts\interpreter.exe' -> 'C:\Python311\Scripts\interpreter.exe.deleteme'
> 
> which indicates that interpreter.exe is not installed in the python scripts folder hence interpreter command won't work.
> 
> **Solution:** Open command prompt with admin rights If you ran `pip install open-interpreter` previously, run `pip uninstall open-interpreter` first. Then `pip install open-interpreter` to install again and no error should show up. Run `interpreter` and voila

It works!!! Thx!!!

**Comment by jorbey** [comment]
> If you see an error similar to the following.
> 
> ```
> interpreter command not found
> ```
> 
> or
> 
> ```
> interpreter : The term 'interpreter' is not recognized as the
> name of acmdlet, function, script file, or operable program
> ```
> 
> This is likely an issue with your Python/pip `Scripts` directory not being in your `PATH` environment variable.
> 
> To find out what that path is, run the following command.
> 
> ```
> python -c 'import os,sysconfig;print(sysconfig.get_path("scripts"))'
> ```
> 
> (If that doesn't work, see my comment below: [#169 (comment)](https://github.com/OpenInterpreter/open-interpreter/issues/169#issuecomment-1713124018))
> 
> Now [add that directory](https://gist.github.com/nex3/c395b2f8fd4b02068be37c961301caa7#file-path-md) to your PATH environment variable and restart your terminal.
> 
> If you still run into problems, leave a comment here or ask on [the Discord server](https://discord.gg/YG7APUyJ5).

Not worked for me, but I fixed.
I am using macOS, python3 is included.
After installed, run "interpreter" in terminal show:interpreter command not found.
I found that "interpreter" is in "/Users/_(acount_name)_/Library/Python/3.9/bin".
Then add that directory to PATH environment variable and restart terminal, It worked!

**Comment by AlvinKimata** [comment]
> If you run the following command, it should tell you where the `interpreter` executable is.
> 
> ```
> pip show --files open-interpreter
> ```
> 
> ```
> Name: open-interpreter
> Version: 0.1.3
> Summary: Let language models run code locally.
> Home-page:
> Author: Killian Lucas
> Author-email: killian@drinkwater.ai
> License:
> Location: c:\users\jordan\scoop\persist\miniconda3\envs\oi\lib\site-packages
> Requires: appdirs, astor, git-python, huggingface-hub, inquirer, litellm, openai, pyreadline3, python-dotenv, rich, six, tiktoken, tokentrim, wget
> Required-by:
> Files:
>   ..\..\Scripts\interpreter.exe
>   interpreter\README.md
>   interpreter\__init__.py
>   ...
> ```
> 
> The first path under the `Files:` list should be the `interpreter` executable. It should be relative to the path to the `pip` executable, which you can find with the following command.
> 
> ```
> # Windows
> where.exe pip
> 
> # Linux
> which pip
> ```
> 
> On my Windows computer, it returns two paths for `pip` because I have Miniconda installed. The first path listed is the one that the `interpreter` path is relative to.
> 
> ```
> C:\Users\Jordan\scoop\apps\miniconda3\current\envs\oi\Scripts\pip.exe
> C:\Users\Jordan\scoop\apps\python\current\Scripts\pip.exe
> ```
> 
> So, resolving `..\..\Scripts\interpreter.exe` relative to `C:\Users\Jordan\scoop\apps\miniconda3\current\envs\oi\Scripts\pip.exe` gives me `C:\Users\Jordan\scoop\apps\miniconda3\current\envs\oi\Scripts\interpreter.exe`
> 
> So, in my case, I need to make sure that `C:\Users\Jordan\scoop\apps\miniconda3\current\envs\oi\Scripts` is in my `PATH` environment variable.

Thanks alot! This worked for me.
